packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -266,38 +265,18 @@ const DropdownMenuRootContent = React.forwardRef<
       {...contentProps}
       ref={forwardedRef}
       portalled={portalled}
-      onCloseAutoFocus={(event) => {
-        props.onCloseAutoFocus?.(event);
-
-        if (!event.defaultPrevented) {
-          if (!hasInteractedOutsideRef.current) context.triggerRef.current?.focus();
-          // Always prevent auto focus because we either focus manually or want user agent focus
-          event.preventDefault();
-        }
-
+      onCloseAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
+        if (!hasInteractedOutsideRef.current) context.triggerRef.current?.focus();
         hasInteractedOutsideRef.current = false;
-      }}
-      onInteractOutside={(event) => {
-        props.onInteractOutside?.(event);
-
-        if (!event.defaultPrevented) {
-          const originalEvent = event.detail.originalEvent as PointerEvent;
-          const ctrlLeftClick = originalEvent.button === 0 && originalEvent.ctrlKey === true;
-          const isRightClick = originalEvent.button === 2 || ctrlLeftClick;
-
-          if (!context.modal || isRightClick) hasInteractedOutsideRef.current = true;
-        }
-
-        // Prevent dismissing when clicking the trigger.
-        // As the trigger is already setup to close, without doing so would
-        // cause it to close and immediately open.
-        //
-        // We use `onInteractOutside` as some browsers also
-        // focus on pointer down, creating the same issue.
-        const target = event.target as HTMLElement;
-        const targetIsTrigger = context.triggerRef.current?.contains(target);
-        if (targetIsTrigger) event.preventDefault();
```

Comment:
Going back to my earlier comment again (and it seems @andy-hook's comment too), I'm not sure this is the way to go.

I believe it comes down to whether we consider the trigger as only opening vs. toggling.
If I imagine a non-modal dropdown menu, or a non-modal popover, and I don't want clicking outside to close (say I disabled it), then, would I still want/expect clicking the trigger to close the menu?

I feel like I would and that's part of it's functionality.